src/PMGDQueryHandler.cc

```diff
@@ -241,6 +244,50 @@ int PMGDQueryHandler::add_node(const protobufs::AddNode &cn,
     return 0;
 }
 
+int PMGDQueryHandler::update_node(const protobufs::UpdateNode &un,
+                    protobufs::CommandResponse *response)
+{
+    long id = un.identifier();
+    bool query = un.has_query_node();
+
+    // Make sure there is either an id or a query node
+    if (id < 0 && !query) {
```

Comment:
Is it also an error if /both/ are set?

Reply:
True. Added a comment on this on the code.